Source/Particles/RigidInjectedParticleContainer.cpp

```diff
@@ -348,7 +348,8 @@ RigidInjectedParticleContainer::Evolve (int lev,
     const Real* plo = Geom(lev).ProbLo();
     const Real* phi = Geom(lev).ProbHi();
     const int zdir = AMREX_SPACEDIM-1;
-    done_injecting[lev] = (zinject_plane_levels[lev] < plo[zdir] || zinject_plane_levels[lev] > phi[zdir]);
+    done_injecting[lev] = ((zinject_plane_levels[lev] < plo[zdir] && WarpX::moving_window_v > 0.) ||
+                           (zinject_plane_levels[lev] > phi[zdir] && WarpX::moving_window_v < 0.));
```

Comment:
Strictly speaking, shouldn't this be something like
```
(zinject_plane_levels[lev] < plo[zdir] && WarpX::moving_window_v > vinject_plane) ||
(zinject_plane_levels[lev] > phi[zdir] && WarpX::moving_window_v < vinject_plane)
```
?
If getting the velocity of injected plane is cumbersome, we can do
```
AMREX_ALWAYS_ASSERT_WITH_MESSAGE(
    WarpX::moving_window_v == PhysConst::c || 
    WarpX::moving_window_v == -PhysConst::c || 
    WarpX::moving_window_v == 0.,
    "Moving window velocity must be -c, 0 or c to use rigid injection");
```

Reply:
Currently, the injection plane assumed to be stationary in the lab frame (i.e. vinject_plane = 0). If that changes, your change could then be implemented. However, what I had isn't correct, since it fails for moving_window_v = 0 and doesn't take into account the boosted frame. I think it needs to be this:

`    done_injecting[lev] = ((zinject_plane_levels[lev] < plo[zdir] && WarpX::moving_window_v + WarpX::beta_boost*PhysConst::c >= 0.) ||
                           (zinject_plane_levels[lev] > phi[zdir] && WarpX::moving_window_v + WarpX::beta_boost*PhysConst::c <= 0.));
`

Reply:
I think that is correct. Do you want to commit it, and maybe add another
```
AMREX_ASSERT(vinject_plane == 0)
```
so that we do not forget to update this part if we change the code later?

Reply:
I'll commit the change.

I don't see vinject_plane defined anywhere.

Reply:
Ok let's not worry about the `ASSERT` for now then